### PR TITLE
Adding message to require functions in 2 contracts #888

### LINKED ISSUE #888: Add reason string for all assertions/requires
The next solc version introduces support for reason strings for error messages. As soon as this release is available, migrate to that version and annotate all `require`s and `assert`s in the OpenZeppelin codebase with meaningful error messages.

- 📈 This is a feature request.

See https://github.com/ethereum/solidity/pull/3364

### PR DESCRIPTION
**Pull request by pmosse**
Fixes #888

Adding explanation message to require functions in the base code. Starting with 2 contracts to discuss the guidelines. Will also include revert functions.

- [X] 📘 I've reviewed the [OpenZeppelin Contributor Guidelines](../blob/master/CONTRIBUTING.md)
- [X] ✅ I've added tests where applicable to test my new functionality.
- [X] 📖 I've made sure that my contracts are well-documented.
- [X] 🎨 I've run the JS/Solidity linters and fixed any issues (`npm run lint:all:fix`).

### CODE REVIEW

File: contracts/payment/SplitPayment.sol [hunk 1]
```diff
@@ -40,13 +40,13 @@ contract SplitPayment {
...
 
-    require(shares[payee] > 0);
+    require(shares[payee] > 0, "No share was found");
 
     uint256 totalReceived = address(this).balance.add(totalReleased);
     uint256 payment = totalReceived.mul(shares[payee]).div(totalShares).sub(released[payee]);
 
-    require(payment != 0);
-    require(address(this).balance >= payment);
+    require(payment != 0, "The share for the payee is zero");
```
Comment (elopio) [inline contracts/payment/SplitPayment.sol]: Wouldn't this be like:
`All shares for sender already claimed`?

File: contracts/payment/SplitPayment.sol [hunk 2]
```diff
@@ -40,13 +40,13 @@ contract SplitPayment {
...
+    require(address(this).balance >= payment, "Not enough funds to do the payment");
```
Comment (elopio) [inline contracts/payment/SplitPayment.sol]: I suggest:
`Not enough funds for payment`

**Comment by pmosse** [comment]
Great suggestions @elopio. Waiting for the discussion in https://github.com/OpenZeppelin/openzeppelin-solidity/issues/888 before moving forward.

**Comment by nventuro** [comment]
Closing - a full-fledged solution will come out of #888.

**Pull request closed without merging**